models/api-key.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+const AWS = require('aws-sdk');
+const crypto = require('crypto');
+const dynamoDb = new AWS.DynamoDB.DocumentClient();
+const response = require ('../helpers/response.js');
+
+module.exports.activate = (requestBody, callback) => {
+  const data = JSON.parse(requestBody);
+  
+  if ((!data.email) || typeof data.email !== 'string') {
+    response.returnError(400, 'email is required', callback);
+    return;
+  }
+  
+  if ((!data.apiKey) || typeof data.apiKey !== 'string') {
+    response.returnError(400, 'apiKey is required', callback);
+    return;
+  }
+  
+  getApiKeyByValue(data.apiKey, (error, apiKeyRecord) => {
+    if (error) {
+      console.error(error);
+      response.returnError(500, 'Failed to retrieve API Key.', callback);
+      return;
+    }
+    
+    const noSuchApiKeyRecord = (apiKeyRecord == undefined);
+    if (noSuchApiKeyRecord || (apiKeyRecord.email !== data.email)) {
+      response.returnError(404, 'No matching API Key record was found', callback);
+      return;
+    }
+    
+    if (isAlreadyActivated(apiKeyRecord)) {
+      console.log(
+        'Attempt to re-activate API Key %s...',
+        data.apiKey.substr(0, data.apiKey.length / 2)
+      );
+      response.returnError(401, 'Unauthorized', callback);
+      return;
+    }
+    
+    const apiSecret = crypto.randomBytes(32).toString('base64');
+    apiKeyRecord.activatedAt = new Date().getTime();
+    updateApiKeyRecord(apiKeyRecord, (error) => {
+      if (error) {
+        console.error(error);
+        response.returnError(500, 'Error while activating API Key.', callback);
+        return;
+      }
+      
+      response.returnSuccess({'apiSecret': apiSecret}, callback);
+      return;
+    });
+  });
+};
+
+module.exports.create = (requestBody, callback) => {
+  const timestamp = new Date().getTime();
+  
+  /* @TODO Make sure we were given a JSON body. */
+  
+  const data = JSON.parse(requestBody);
+  
+  if ((!data.email) || typeof data.email !== 'string') {
+    response.returnError(400, 'email is required', callback);
+    return;
+  }
+  
+  /* @todo Validate that it seems to be an actual email address. */
+  
+  const apiKeyValue = crypto.randomBytes(20).toString('hex');
+  const params = {
+    TableName: process.env.TABLE_NAME,
+    Item: {
+      createdAt: timestamp,
+      email: data.email,
+      value: apiKeyValue
+    }
+  };
+  
+  dynamoDb.put(params, (error, result) => {
+    if (error) {
+      console.error(error);
+      response.returnError(500, 'Failed to save new API Key.', callback);
+      return;
+    }
+    
+    response.returnSuccess(null, callback);
+    return;
+  });
+};
+
+const getApiKeyByValue = (value, callback) => {
```

Comment:
Since I'm not calling any functions / doing anything (other than requiring dependencies and defining functions) at the root level of this file, I don't think that will cause any problems. To trigger that error, I believe it would require calling `getApiKeyByValue()` at the root level of this file (i.e. - not from within a function) and before line 94.